celery/backends/database/models.py

```diff
@@ -10,14 +10,16 @@
 
 __all__ = ('Task', 'TaskExtended', 'TaskSet')
 
+DialectSpecificInteger = sa.Integer().with_variant(sa.BigInteger, 'mssql')
+
 
 class Task(ResultModelBase):
     """Task result/status."""
 
     __tablename__ = 'celery_taskmeta'
     __table_args__ = {'sqlite_autoincrement': True}
 
-    id = sa.Column(sa.Integer, sa.Sequence('task_id_sequence'),
+    id = sa.Column(DialectSpecificInteger, sa.Sequence('task_id_sequence'),
```

Comment:
is this change backward compatible?

Reply:
I’m not sure if it’s elegant enough, but see the next commit.

I also want to mention that the change already partially covered by existing tests: SQLite won’t allow anything other than *integer* for the auto-increment PK column.